.github/workflows/e2e.yaml

```diff
@@ -80,10 +80,14 @@ jobs:
       - name: Run user impersonation tests
         run: |
           kubectl -n kustomize-system patch deploy/kustomize-controller -p "$(cat ./config/testdata/user-impersonation/patch.yaml)"
-          kubectl -n tenant-a apply -f ./config/testdata/user-impersonation/test.yaml
+          kubectl -n tenant-a apply -f ./config/testdata/user-impersonation/use-test.yaml
           kubectl -n tenant-a wait kustomizations/podinfo --for=condition=ready --timeout=4m
-          kubectl -n tenant-b apply -f ./config/testdata/user-impersonation/test2.yaml
+          kubectl -n tenant-a  delete kustomizations/podinfo --wait
+          kubectl -n tenant-b apply -f ./config/testdata/user-impersonation/sa-test.yaml
           kubectl -n tenant-b wait kustomizations/podinfo --for=condition=ready --timeout=4m
+          kubectl -n tenant-b  delete kustomizations/podinfo --wait
+          kubectl -n tenant-c apply -f ./config/testdata/user-impersonation/fail-sa-test.yaml
+          until kubectl -n kustomize-system logs deploy/kustomize-controller 2>&1 | grep "Error from server (Forbidden)" ; do sleep 2; done
```

Comment:
Replace the logs search with `kubectl -n tenant-c get kustomization podinfo -oyaml`, as the `Forbidden` error should be in the Kustomization status message.